public/app/plugins/datasource/elasticsearch/configuration/ConfigEditor.tsx

```diff
@@ -57,6 +57,10 @@ export const ConfigEditor = (props: Props) => {
         renderSigV4Editor={<SIGV4ConnectionConfig {...props}></SIGV4ConnectionConfig>}
       />
 
+      {config.featureToggles.secureSocksDatasourceProxy && (
```

Comment:
Did you consider adding this to the `DataSourceHttpSettings` component instead so that you don't have to update all datasource config editors? Each datasource could provide a prop whether it should be enabled or not (and feature toggle checked within the ``DataSourceHttpSettings``.

Reply:
I did consider this, but in the next month, we'll be adding support for other, non http-protocol based datasources, so this way, it'll be easier to iterate on in the future.

Reply:
Understood